### PR TITLE
PAINTROID-309 Refactor DefaultFillToolOptionsView to Kotlin

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
File that was refactored: DefaultFillToolOptionsView
Replaced `SeekBar` and `EditText` with AndroidX equivalents.

https://jira.catrob.at/browse/PAINTROID-309

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+                if (fromUser) {
+                    setColorToleranceText(progress)
+                }
+            }
+
+            override fun onStartTrackingTouch(seekBar: SeekBar) = Unit
+            override fun onStopTrackingTouch(seekBar: SeekBar) = Unit
+        })
+        colorToleranceEditText.addTextChangedListener(object : TextWatcher {
```
Comment (FromG) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt:62]: new line would be appreciated afer line 61

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) =
+                Unit
+
+            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) = Unit
```
Comment (FromG) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt:66]: also here after line 66

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            override fun afterTextChanged(s: Editable) {
+                try {
+                    val colorToleranceInPercent = s.toString().toInt()
+                    colorToleranceSeekBar.progress = colorToleranceInPercent
+                    updateColorTolerance(colorToleranceInPercent)
+                } catch (e: NumberFormatException) {
+                    Log.e("Error parsing tolerance", "result was null")
+                }
+            }
+        })
```
Comment (FromG) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt:76]: also after line 76.

**Review by FromG — CHANGES_REQUESTED** [review]
Rest looks good except new line and the static analysis warning in jenkins.

**Comment by CodeChamp-SS** [comment]
> Rest looks good except new line and the static analysis warning in jenkins.

done :)

**Review by FromG — APPROVED** [review]
lgtm

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        callback?.onColorToleranceChanged(colorTolerance)
+    }
+
+    private fun setColorToleranceText(toleranceInPercent: Int) {
+        colorToleranceEditText.setText(String.format(Locale.getDefault(), "%d", toleranceInPercent))
+    }
+
+    override fun setCallback(callback: FillToolOptionsView.Callback) {
+        this.callback = callback
+    }
```
Comment (gPathpp) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt:98]: Can this maybe be done with [at]override callback?

Comment (CodeChamp-SS) [reply Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultFillToolOptionsView.kt:98]: idts 🤔

**Review by gPathpp — APPROVED** [review]
LGTM

**Pull request merged**